src/main/java/org/prebid/server/bidder/consumable/ConsumableBidder.java

```diff
@@ -1,227 +1,261 @@
-        if (usPrivacy != null) {
-            requestBuilder.usPrivacy(usPrivacy);
+    private ExtImpConsumable parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), CONS_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage());
         }
+    }
 
-        final Integer gdpr = extRegs != null ? extRegs.getGdpr() : null;
-        final User user = request.getUser();
-        final ExtUser extUser = user != null ? user.getExt() : null;
-        final String gdprConsent = extUser != null ? extUser.getConsent() : null;
-        if (gdpr != null || gdprConsent != null) {
-            final ConsumableBidGdpr.ConsumableBidGdprBuilder bidGdprBuilder = ConsumableBidGdpr.builder();
-            if (gdpr != null) {
-                bidGdprBuilder.applies(gdpr != 0);
-            }
-            if (gdprConsent != null) {
-                bidGdprBuilder.consent(gdprConsent).build();
-            }
-            requestBuilder.gdpr(bidGdprBuilder.build());
+    private boolean isImpValid(Site site, App app, ExtImpConsumable impExt) {
+        return (app != null && !Strings.isNullOrEmpty(impExt.getPlacementId()))
+                || (site != null && impExt.getSiteId() != 0 && impExt.getNetworkId() != 0 && impExt.getUnitId() != 0);
+
+    }
+
+    private Imp modifyImp(Imp imp, ExtImpConsumable impExt) {
+        final UpdateResult<ObjectNode> impExtUpdateResult = modifyImpExt(imp, impExt);
+
+        return imp;
+    }
+
+    private UpdateResult<ObjectNode> modifyImpExt(Imp imp, ExtImpConsumable extImpConsumable) {
+        final Integer siteId = extImpConsumable.getSiteId();
+        final String unitName = extImpConsumable.getUnitName();
+        final Integer unitId = extImpConsumable.getUnitId();
+        final Integer networkId = extImpConsumable.getNetworkId();
+        final String placementId = extImpConsumable.getPlacementId();
+        final ObjectNode impExt = imp.getExt();
+
+        final ObjectNode updatedExt = impExt.deepCopy();
+        if (Strings.isNullOrEmpty(placementId)) {
+            updatedExt.set("siteId", TextNode.valueOf(Integer.toString(siteId)));
+            updatedExt.set("unitName", TextNode.valueOf(unitName));
+            updatedExt.set("unitId", TextNode.valueOf(Integer.toString(unitId)));
+            updatedExt.set("networkId", TextNode.valueOf(Integer.toString(networkId)));
+        } else {
+            updatedExt.set("placementId", TextNode.valueOf(placementId));
         }
 
+        return UpdateResult.updated(updatedExt);
+    }
+
+    private BidRequest modifyBidRequest(BidRequest bidRequest, List<Imp> imps) {
+        return bidRequest.toBuilder().imp(imps).build();
+    }
+
+    private String constructUri(String placementId) {
+        final String uri = Strings.isNullOrEmpty(placementId) ? SITE_URI_PATH : (APP_URI_PATH + placementId);
+        return this.endpointUrl + uri;
+    }
```

Comment:
the URIs are not checked in the unit tests

Reply:
Done